### PR TITLE
Handbrake 1.1.1 update

### LINKED ISSUE #2903: Handbrake: update FFmpeg dependency
Tested on hrev52201 x86_64, handbrake won't install due to older FFmpeg dependency.

### PR DESCRIPTION
**Pull request by kenmays**
Handbrake v1.1.1 released on Monday, June 18, 2018
See: https://handbrake.fr/news.php
Docs: https://handbrake.fr/docs/en/1.1.0/

Note: closes #2903

### CODE REVIEW

**Comment by kenmays** [comment]
Resolved issue and updated recipe.

File: media-video/handbrake/handbrake-1.1.1.recipe [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+# of the current state of the HandBrake port
+
+PATCH()
+{
+	sed -i 's/MODULES += contrib\/.*//g' make/include/main.defs
+}
+
+BUILD()
+{
+#	./configure --prefix=$prefix --force --disable-x265
```
Comment (extrowerk) [inline media-video/handbrake/handbrake-1.1.1.recipe]: Are you sure it isn't needed?

**Comment by kenmays** [comment]
Refreshed recipe.

File: media-video/handbrake/handbrake-1.1.1.recipe [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+	sed -i 's/MODULES += contrib\/.*//g' make/include/main.defs
+}
+
+BUILD()
+{
+	./configure --prefix=$prefix --force --disable-x265
+#	cd build
+#	make
+#	cd ../haiku
+#	make
```
Comment (extrowerk) [inline media-video/handbrake/handbrake-1.1.1.recipe]: You have to uncomment this line (113), and delete the commented lines (110,111,112)

File: media-video/handbrake/handbrake-1.1.1.recipe [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+INSTALL()
+{
+#	cd build
+#	make install
```
Comment (extrowerk) [inline media-video/handbrake/handbrake-1.1.1.recipe]: Uncomment this

Comment (extrowerk) [reply media-video/handbrake/handbrake-1.1.1.recipe]: And delete every commented line around (118, 120, 121, 122)

File: media-video/handbrake/handbrake-1.1.1.recipe [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+PATCHES="handbrake-$portVersion.patchset"
+ADDITIONAL_FILES="HandBrake.rdef"
+
+ARCHITECTURES="!x86_gcc2 ?x86 x86_64"
+SECONDARY_ARCHITECTURES="?x86"
+
+PROVIDES="
+	handbrake$secondaryArchSuffix = $portVersion
+	cmd:HandBrakeCLI
+#	app:HandBrake
```
Comment (extrowerk) [inline media-video/handbrake/handbrake-1.1.1.recipe]: Delete this commented line

File: media-video/handbrake/handbrake-1.1.1.recipe [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+	local MAJOR="`echo "$portVersion" | cut -d. -f1`"
+	local MIDDLE="`echo "$portVersion" | cut -d. -f2`"
+	local MINOR="`echo "$portVersion" | cut -d. -f3`"
+	sed \
+		-e "s|@MAJOR@|$MAJOR|" \
+		-e "s|@MIDDLE@|$MIDDLE|" \
+		-e "s|@MINOR@|$MINOR|" \
+		$portDir/additional-files/HandBrake.rdef > HandBrake.rdef
+#
+#	addResourcesToBinaries HandBrake.rdef $prefix/bin/HandBrakeCLI
```
Comment (extrowerk) [inline media-video/handbrake/handbrake-1.1.1.recipe]: Uncomment this, delete others.

File: media-video/handbrake/handbrake-1.1.1.recipe [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+}
+
+BUILD()
+{
+	./configure --prefix=$prefix --force --disable-x265
+	make
+}
+
+INSTALL()
+{
```
Comment (extrowerk) [inline media-video/handbrake/handbrake-1.1.1.recipe]: You mistakenly removed the ```make install```. Please add it back.

File: media-video/handbrake/handbrake-1.1.1.recipe [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+PATCH()
+{
+	sed -i 's/MODULES += contrib\/.*//g' make/include/main.defs
+}
+
+BUILD()
+{
+	./configure --prefix=$prefix --force --disable-x265
+	make
```
Comment (extrowerk) [inline media-video/handbrake/handbrake-1.1.1.recipe]: cant we add $jobArgs to make? Deos it breaks with parallel jobs?

**Comment by extrowerk** [comment]
Looks good, but intested by me.

File: media-video/handbrake/patches/handbrake-1.1.1.patchset [hunk 8]
```diff
@@ -0,0 +1,3839 @@
...
++ScanWindow::ScanWindow( hb_handle_t * handle )
++    : BWindow( BRect( 0,0,10,10 ), "Scan", B_FLOATING_WINDOW_LOOK,
++               B_MODAL_SUBSET_WINDOW_FEEL, B_NOT_CLOSABLE |
++               B_NOT_RESIZABLE | B_NOT_ZOOMABLE )
++{
++    /* Add the scan view */
++    fView = new ScanView( handle );
++    AddChild( fView );
++    
++    /* Resize to fit
```
Comment (korli) [inline media-video/handbrake/patches/handbrake-1.1.1.patchset]: This looks wrong: why add a second " x86_64:Haiku:*:*" ?

**Comment by kenmays** [comment]
@korli - Removed 1.0.7 recipe

**Comment by fbrosson** [comment]
@kenmays, I guess you already noticed that HandBrake 1.1.2 was released yesterday.

* https://handbrake.fr/news.php?article=41
* https://github.com/HandBrake/HandBrake/releases/tag/1.1.2

BTW, if you don't need to close this PR and open a new one for `1.1.2` if you wish to switch to `1.1.2`. Indeed, you can rename the title of this PR and update it. BTW, if/when you update your recipe to `1.1.2` you might wish to rebase it against master. (Of course, you'll need to `git push --force` if you choose to rebase.)

It doesn't matter that the name you chose for the branch is `handbrake-1.1.1`. BTW, we generally don't include the version in the name of the branch.
(Well, some users do, but that's because they don't care about deleting their old branches :smile:)

**Comment by scottmc** [comment]
Patch(es) for this should be sent upstream as one or more pull requests, so that we don't have to keep matching the same patches on each new version.

**Pull request merged**